### PR TITLE
sdcardCheck: minor refactor

### PR DESCRIPTION
**Pull request by MaEtUgR**
**Describe problem solved by this pull request**
I was late for the review in https://github.com/PX4/PX4-Autopilot/pull/16707. Making up for that.

**Describe your solution**
- No unnecessary stack allocation when not doing the actual sdcard check
- No early return
- parameter variable names after convention (I know I forgot this myself before)
- Very explicit check if sd card is present

**Test data / coverage**
So far untested

### CODE REVIEW

File: src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp [hunk 1]
```diff
@@ -43,30 +44,29 @@
...
 
-	if (total_bytes == 0) { // on NuttX we get 0 total bytes if no SD card is inserted
+	if (param_com_arm_sdcard > 0) {
+		bool sdcard_present{false};
+		struct statfs statfs_buf;
 
-		if (reporting_enabled == 2) {
-			success = false;
+		if (statfs(PX4_STORAGEDIR, &statfs_buf) == 0) {
+			// on NuttX we get 0 total bytes = f_blocks * f_bsize if no SD card is inserted
```
Comment (cmic0) [inline src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp:57]: The comment has to be adapted

**Review by cmic0 — DISMISSED** [review]
just a minor detail on the comment otherwise looks good

File: src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp (hunk 1 above)
Comment (MaEtUgR) [reply src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp:57]: I did but probably I should make it unambiguous.

Comment (MaEtUgR) [reply src/modules/commander/Arming/PreFlightCheck/checks/sdcardCheck.cpp]: I made it more clear, what do you think?

**Review by dagar — APPROVED** [review]

**Pull request merged**